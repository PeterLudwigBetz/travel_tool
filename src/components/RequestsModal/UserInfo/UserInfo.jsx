
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ImageLink from '../../image-link/ImageLink';

class UserInfo extends Component {
  render() {
    const { requestData, user } = this.props;
    return (
      <div>
        <div className="modal__user-info">
          <ImageLink
            imageSrc={user.picture}
            altText="avatar"
            imageClass="modal__oval"
          />
          <span className="modal__text-size">
            {requestData && requestData.name}
          </span>
          <div className="modal__modal3">
            {requestData && requestData.role}
,
            {requestData && requestData.department}
          </div>
        </div>
      </div>
    );
  }
}

UserInfo.propTypes = {
  requestData: PropTypes.object,
  user: PropTypes.object,
};

UserInfo.defaultProps = {
  requestData: {},
  user: {},
};

export default UserInfo;
