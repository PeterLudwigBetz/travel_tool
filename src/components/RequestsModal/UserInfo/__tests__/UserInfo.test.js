import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from '../UserInfo';
import requestData from '../../__mocks__/requestData';

const props = {
  user: {
    picture: 'http://my-image'
  },
  requestData,
};

describe('UserComments component', () => {
  const wrapper = shallow(<UserInfo {...props} />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the UserComments component as expected', () => {
    expect(wrapper.find('div').length).toBe(3);
    expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find('ImageLink').length).toBe(1);
    expect(wrapper.length).toBe(1);
  });
});
