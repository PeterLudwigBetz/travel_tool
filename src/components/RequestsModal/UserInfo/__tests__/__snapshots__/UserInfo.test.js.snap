// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComments component should match snapshot 1`] = `
<div>
  <div
    className="modal__user-info"
  >
    <ImageLink
      altText="avatar"
      imageClass="modal__oval"
      imageSrc="http://my-image"
    />
    <span
      className="modal__text-size"
    >
      Test user C
    </span>
    <div
      className="modal__modal3"
    >
      Software Developer
      ,
      TDD
    </div>
  </div>
</div>
`;
